Newsletters: add 157 (2021-07-14)

harding opened the pull request:
- [x] Update releases/RCs @harding 
- [x] Add topic links @harding
- [x] Bitcoin Core PR Review Club meeting summary @glozow 
- [x] `C-Lightning #4625` offers update @dongcarl 
- [x] `Eclair #1746` postgres support @Xekyo 
- [x] `BIPs #943` bip-anyprevout @adamjonas

Note: the news section links to a mailing list post of mine; extra scrutiny of that item welcome!

glozow commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,204 @@
+---
+title: 'Bitcoin Optech Newsletter #156'
```
btw

```suggestion
title: 'Bitcoin Optech Newsletter #157'
```

harding replied:
Yikes, thanks!

bitschmidty commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,208 @@
+    `OP_CAT` also makes it possible to create recursive
+    [covenants][topic covenants] which could permanently restrict the
+    spendability of any bitcoins committed to the covenant.  Some people
+    have objected to allowing covenants in Bitcoin, but several
```
Are there any objections from the thread that help inform the debate for the reader that could be linked here?

bitschmidty commented on _includes/specials/taproot/en/03-p2wpkh-to-p2tr.md:
```diff
@@ -0,0 +1,119 @@
+For wallets that already support receiving and spending v0 segwit P2WPKH
+outputs, upgrading to v1 segwit P2TR for single sig should be easy.
```
s/single sig/single-sig ?

bitschmidty commented on _includes/specials/taproot/en/03-p2wpkh-to-p2tr.md:
```diff
@@ -0,0 +1,119 @@
+outputs, upgrading to v1 segwit P2TR for single sig should be easy.
+Here are the main steps:
+
+- **Use a new BIP32 key derivation path:** you don't need to change your
```
I missed this in the other 3 of this series, but we lose the bulleted list anchor links when embedding

harding replied:
Good catch!  I'm not sure why it doesn't work and I can't think of a trivial fix for it, so I opened a new issue.  I'll look into it ASAP.

jnewbery replied:
Marking this as resolved now that we have an issue tracking it.

bitschmidty commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,208 @@
+- [BIPs #943][] ajtowns/bip-anyprevout FIXME:adamjonas
+
+- [BTCPay Server #2655][] signals to web browsers that they should not
+  send the HTTP `referer` <!-- sic --> field when the user clicks on an
```
s/an link/a link

jnewbery replied:
fixed

bitschmidty commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,208 @@
+
+- [BTCPay Server #2655][] signals to web browsers that they should not
+  send the HTTP `referer` <!-- sic --> field when the user clicks on an
+  link to a transaction explorer.  This avoids telling the block
```
Suggest adding link to block explorer topic.

jnewbery replied:
fixed

glozow commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,207 @@
+`script_to_p2wsh_script` in script\_util.py do?"
+  a0="These are helper functions to construct `CScript` objects for P2PKH, P2SH, P2WPKH, and P2WSH
+scripts from public keys and scripts."
+  a0link="https://bitcoincore.reviews/22363#31-90"
```
Oops, thanks! Confused the `l` with a 1

glozow commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,207 @@
+transaction, the script interpreter recognizes a witness program to select which scriptCode to use in
+script validation. P2SH and P2WSH output types commit to a script hash in the scriptPubKey. The
+spender must provide the redeemScript and/or witness script that matches the hash, as well as the
+data to satisfy the script itself."
```
Gotcha, I split it up!

jnewbery commented on _posts/en/newsletters/2021-07-14-newsletter.md:
```diff
@@ -0,0 +1,223 @@
+scriptSig and/or witness."
+  a1link="https://bitcoincore.reviews/22363#l-31"
+
+  q2="Define witness program and scriptCode."
```
I think `scriptCode` is quite confusing and is used to mean more than one thing. I suggest you remove this question/answer.

jnewbery commented on _includes/specials/taproot/en/03-p2wpkh-to-p2tr.md:
```diff
@@ -0,0 +1,120 @@
+  addresses like P2WPKH.
+
+- **Creating a spending transaction:** all the non-witness fields for
+  taproot are the same as for P2WPKH.  The only difference for a
+  single-sig P2TR is that you'll only have one witness element (a
+  signature) instead of the two for P2WPKH (a key and a signature).
```
It seems a bit premature to be talking about the witness here, when it only gets computed in the next two steps.

harding commented:
Pushed edits for @jnewbery feedback in my section (thanks!), reviewed and made some small edits to sections contributed by @adamjonas, @dongcarl, @glozow, and @Xekyo (thanks!), updated releales/RC, and added topic links.

Thanks everyone!

jnewbery commented:
Made some very minor changes to the pr review club summary. Everything else LGTM. Thanks team!

jnewbery commented on _includes/specials/taproot/en/03-p2wpkh-to-p2tr.md:
```diff
@@ -0,0 +1,119 @@
+uses][wiki taproot uses] and [bech32m adoption][wiki bech32 adoption]
+pages of the Bitcoin Wiki so other developers can learn from your code.
+
+[^electrum-segwit]:
```
How wallets can handle different script types seems like too much good/important information to be a footnote. Perhaps we can turn this into its own post at some point in the future?

harding replied:
Sure, although I don't think there's much more to say than what's here[1].  Were you thinking that's something we'd put into this taproot series, or did you have some other idea for where we'd put it?

[1] Or so I say now.  When I actually write the thing and I'm apologizing for how long it is, you can point back here and laugh at me.  :-)

jnewbery commented on _topics/en/sighash_noinput.md:
```diff
@@ -78,6 +78,9 @@ optech_mentions:
   - title: "Using schnorr signatures plus `OP_CAT` to simulate `SIGHASH_ANYPREVOUT`"
     url: /en/newsletters/2021/02/03/#replicating-op-checksigfromstack-with-bip340-and-op-cat
 
+  - title: "BIPs #943 updates BIP118 to provide SIGHASH_ANYPREVOUT rather than SIGHASH_NOINPUT"
```
"provide" doesn't seem like the right verb here. "updates BIP118 to be named SIGHASH_ANYPREVOUT rather than SIGHASH_NOINPUT"?

bitschmidty replied:
Took this suggestion

bitschmidty commented:
Squashed and merged! Thanks to all contributors this week @harding @glozow @dongcarl @Xekyo @adamjonas + 👁️‍🗨️  @jnewbery
